### PR TITLE
[Framework][Assets] Set assets manifest strict by default on debug

### PR DESCRIPTION
**Pull request by GromNaN**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

`assets.strict_mode` make it easier to debug when an asset is missing from the manifest. Using `kernel.debug` as default value would release developers from configuring this option (https://github.com/symfony/demo/pull/1168).

This is consistent with Twig environment that is strict on debug by default.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 1]
```diff
@@ -79,7 +79,7 @@
             ->args([
                 abstract_arg('manifest path'),
                 service('http_client')->nullOnInvalid(),
-                false,
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:82]: This default value was discussed with @weaverryan in https://github.com/symfony/symfony/pull/38495#discussion_r505526414

**Review by weaverryan — COMMENTED** [review]
Hmm, this may makes sense. However, it's a BC break: on 5.4 if I have a path that isn't included, then on 6.0, I'll suddenly have an error. Also, there may be people who are putting some of their files directly `public/` and, for those files, they won't be in `manifest.json` and that's ok. I think we probably can't make this change

**Comment by GromNaN** [comment]
That's a good point. People may rely on this fallback behaviour to mix assets listed and not listed in the manifest.json, using a single asset package. Should it be considered as a good practice? This use-case will still be supported, but require the option strict_mode to be set to false explicitly.

I could add a deprecation in 5.4?

**Comment by weaverryan** [comment]
Hmm, the question is: is strict mode such a better default that it’s worth the pain of deprecating it and forcing people to turn it to true. I’m not convinced that it is. So I’m 👎  for making this the default.

**Comment by stof** [comment]
In particular, I don't think that the strict mode should be related to `kernel.debug`, as it is about whether your manifest is exhaustive or no.

**Pull request closed without merging**